test/react/hooks/use-room.test.tsx

```diff
@@ -0,0 +1,295 @@
+import { ChatClient, RoomLifecycle, RoomOptionsDefaults, RoomStatusListener } from '@ably/chat';
+import { act, render, renderHook } from '@testing-library/react';
+import * as Ably from 'ably';
+import React from 'react';
+import { describe, expect, it, vi } from 'vitest';
+
+import { RoomProvider, useRoom, UseRoomResponse } from '../../../src/react/index.ts';
+import { ChatClientProvider } from '../../../src/react/providers/chat-client-provider.tsx';
+import { newChatClient as newChatClientLib } from '../../helper/chat.ts';
+import { randomRoomId } from '../../helper/identifier.ts';
+
+const TestComponent: React.FC<{ callback?: (room: UseRoomResponse) => void }> = ({ callback }) => {
+  const response = useRoom();
+  if (callback) callback(response);
+  return <></>;
+};
+
+vi.mock('ably');
+
+function newChatClient() {
+  return newChatClientLib() as unknown as ChatClient;
+}
+
+describe('useRoom', () => {
+  it('it should throw an error if used outside of RoomProvider', () => {
+    expect(() => render(<TestComponent />)).toThrowErrorInfo({
+      code: 40000,
+      message: 'useRoom hook must be used within a chat RoomProvider',
+    });
+  });
+
+  it('it should get the room from the context without error', () => {
+    const chatClient = newChatClient();
+    let called = false;
+    const roomId = randomRoomId();
+    const TestProvider = () => (
+      <ChatClientProvider client={chatClient}>
+        <RoomProvider
+          id={roomId}
+          attach={false}
+          release={false}
+        >
+          <TestComponent
+            callback={(response) => {
+              expect(response.room.roomId).toBe(roomId);
+              expect(response.attach).toBeTruthy();
+              expect(response.detach).toBeTruthy();
+              expect(response.roomStatus).toBe(RoomLifecycle.Initialized);
+              called = true;
+            }}
+          />
+        </RoomProvider>
+      </ChatClientProvider>
+    );
+    render(<TestProvider />);
+    expect(called).toBe(true);
+  });
+
+  it('should return working shortcuts for attach and detach functions', () => {
```

Comment:
Tiny thing, can we have all tests began the same please, i.e. `it should`... ? :)